### ISSUE TITLE
closed issues should always be moved to the milestone in which they will be released

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#2161]
There are many closed issues that are clearly in the wrong milestone

![image](https://user-images.githubusercontent.com/6306961/213189880-5f1c9e31-16bd-4531-b6c3-d57260028e41.png)

**Comment by tomholub** [comment]
This is the mechanism that I use for other repositories:
 - closed issues that were not taken action on always go to "discussion"
 - when I'm doing a release, I move all newly closed issues from other milestones to the milestone I'm releasing
 - no issues that don't have milestone

Then it's always easy.

Browser extension right now:

![image](https://user-images.githubusercontent.com/6306961/213190342-881232ea-91ab-420e-98d0-b3f32d8996d8.png)


iOS right now

![image](https://user-images.githubusercontent.com/6306961/213190457-7375fe7f-5fb7-48c0-8632-3be71bca6d6d.png)

As it's done now on Android, I bet there are many issues in "later" and "soon" and so on that actually received a PR but were wrongly left in their original milestone.

By having a habit of keeping all except currently smallest milestone clear of finished issues, when you finish one of the issues in those other milestones, they will stand out on the milestone page. Then you can easily move them when you are creating a release.

**Comment by tomholub** [comment]
While on Android you can see that finished issues can easily get lost in other milestones

![image](https://user-images.githubusercontent.com/6306961/213190903-66504819-5588-4bb5-b8e3-86f3f758e2c2.png)


@sosnovsky fyi on milestone management

**Comment by tomholub** [comment]
one more example

![image](https://user-images.githubusercontent.com/6306961/213192252-7610656d-7080-427e-b9d1-dd09732ad369.png)

**Comment by DenBond7** [comment]
Got it. I will fix it. It is strange why it happened.

For example, this one https://github.com/FlowCrypt/flowcrypt-android/issues/1943

![image](https://user-images.githubusercontent.com/2863246/213194183-651c437a-9f9a-4588-9a5c-b9633bf68b74.png)

I don't see `1.5.0` in the history. it is unclear how and when this issue was moved to `1.5.0`

**Comment by tomholub** [comment]
The milestone itself was likely renamed

**Comment by DenBond7** [comment]
Agree. It can be true.

**Comment by DenBond7** [comment]
After reviewing 

![image](https://user-images.githubusercontent.com/2863246/213697096-bc89ca21-a446-4ed9-a663-50e2c7f8de0a.png)

**Comment by DenBond7** [comment]
@tomholub I also propose to update [dependabot.yml](https://github.com/FlowCrypt/flowcrypt-android/blob/master/.github/dependabot.yml) to automatically move dependabot's PRs to a predefined milestone. After that, a milestone also will include dependencies updates.

https://github.com/dependabot/feedback/issues/561
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#milestone

```
# Specify a milestone for pull requests

version: 2
updates:
  - package-ecosystem: "npm"
    directory: "/"
    schedule:
      interval: "weekly"
    # Associate pull requests with milestone "4"
    milestone: 4
```

**Comment by tomholub** [comment]
I suppose it's not wrong. It's up to you. For the rest of the repos, we keep dependabot PRs out of the milestone, because it's very rare we need to check back which milestone it was included in. If needed, we check the commits on master.

In fact, the policy on other repos is to not include any PRs in milestones. The issues to work on are already represented in the issues in the milestone, and PR only gets added to milestone if it didn't have any issue linked. But that also is up to you, as long as you are consistent.

**Comment by DenBond7** [comment]
Got it. It will be better to have the same behavior over the whole repos. For better navigation. So a milestone should include only issues, right?

**Comment by tomholub** [comment]
Correct, except when you make a PR that doesn't have any issue associated. Then PR goes into the milestone in place of the issue that would normally go.

The advantage is, since there are no duplicates (issue + PR) in the milestone, it's easier to compose release notes.

**Comment by DenBond7** [comment]
> Correct, except when you make a PR that doesn't have any issue associated. 

Got it. Like PRs from external collaborators.

**Comment by tomholub** [comment]
Yes, or any time doing a quick fix without associated issue.
